packages/editor/src/core/all-plugins.tsx

```diff
@@ -0,0 +1,612 @@
+import IconBlanksDragAndDrop from '@editor/editor-ui/assets/plugin-icons/icon-blanks-dnd.svg?raw'
+import IconBlanksTyping from '@editor/editor-ui/assets/plugin-icons/icon-blanks-typing.svg?raw'
+import IconBox from '@editor/editor-ui/assets/plugin-icons/icon-box.svg?raw'
+import IconDropzones from '@editor/editor-ui/assets/plugin-icons/icon-dropzones.svg?raw'
+import IconEquation from '@editor/editor-ui/assets/plugin-icons/icon-equation.svg?raw'
+import IconGeogebra from '@editor/editor-ui/assets/plugin-icons/icon-geogebra.svg?raw'
+import IconH5p from '@editor/editor-ui/assets/plugin-icons/icon-h5p.svg?raw'
+import IconHighlight from '@editor/editor-ui/assets/plugin-icons/icon-highlight.svg?raw'
+import IconImage from '@editor/editor-ui/assets/plugin-icons/icon-image.svg?raw'
+import IconInjection from '@editor/editor-ui/assets/plugin-icons/icon-injection.svg?raw'
+import IconTextArea from '@editor/editor-ui/assets/plugin-icons/icon-input-exercise.svg?raw'
+import IconMultimedia from '@editor/editor-ui/assets/plugin-icons/icon-multimedia.svg?raw'
+import IconSpoiler from '@editor/editor-ui/assets/plugin-icons/icon-spoiler.svg?raw'
+import IconTable from '@editor/editor-ui/assets/plugin-icons/icon-table.svg?raw'
+import IconText from '@editor/editor-ui/assets/plugin-icons/icon-text.svg?raw'
+import IconVideo from '@editor/editor-ui/assets/plugin-icons/icon-video.svg?raw'
+import { EditorPluginType } from '@editor/types/editor-plugin-type'
+
+import { loggedInData as loggedInDataDe } from '@/data/de'
+import { loggedInData as loggedInDataEn } from '@/data/en'
+
+// Need this tiny interface, because the mergeDeepObject seems to
+// screw up the types of our i18n strings
+type PluginStrings = Record<
+  EditorPluginType,
+  {
+    title: string
+    description: string
+  }
+>
+
+const germanPluginStrings = loggedInDataDe.strings.editor
+  .plugins as unknown as PluginStrings
+
+const englishPluginStrings = loggedInDataEn.strings.editor
+  .plugins as unknown as PluginStrings
+
+const getPluginNameAndDescription = (
+  locale: 'de' | 'en',
+  pluginType: EditorPluginType
+) => {
+  let name: string
+  let description: string
+  if (locale === 'de') {
+    name = germanPluginStrings[pluginType]?.title
+    description = germanPluginStrings[pluginType]?.description
+  } else if (locale === 'en') {
+    name = englishPluginStrings[pluginType]?.title
+    description = englishPluginStrings[pluginType]?.description
+  } else {
+    throw new Error('Invalid locale')
+  }
+
+  if (!name || !description) {
+    throw new Error(
+      'Missing plugin name or description for plugin type' + pluginType
+    )
+  }
+
+  return {
+    name,
+    description,
+  }
+}
+
+const getInternationalizedPluginStrings = (type: EditorPluginType) => ({
+  de: getPluginNameAndDescription('de', type),
+  en: getPluginNameAndDescription('en', type),
+})
+
+export const AllPlugins = {
```

Comment:
I had the same discussion a few days ago with Lars. 
If we make the editor states explicit here we have to make sure we keep them up to date in this file. 
If we initialise a box plugin with `{plugin: EditorPluginType.Box}` instead of the complete explicit initialState the responsibility to create a valid initialState stays inside the plugin.
Imho this is easier to maintain because it keeps up to date with changes in the plugin.

To be honest I still don't fully understand the use case so maybe this would not work here, but that's what I would prefer if possible.

Reply:
@elbotho We have two issues to solve:

* Some plugins like `blanksExercise` need to be wrapped inside a `exercise` plugin.
* Some plugins like `blanksExercise` are two plugins we want to have in external menus to include the plugin, like one with mode `typing` (fill-in-gaps exercise) and one with the mode `drag-and-drop`) (drag-and-drop words exercise).

Is the a simpler solution for this?

Reply:
@kulla your custom cases make sense. I made an example suggestion for all other examples [here](https://github.com/serlo/frontend/pull/4077/files#r1750605297).